util/upload-instance-slot-map.py

```diff
@@ -26,26 +26,82 @@
 from botocore.exceptions import ClientError
 
 
-def dump_instances(instance_details):
-    with open(instance_details) as data_file:
-        data = json.load(data_file)
+def dump_instances(instance_details, region, aws_credentials=None):
+    """
+    Get instance information from pricing file and dump to instances.json
+    """
+    # Added exception block to check if pricing information is a valid JSON file
+    try:
+        with open(instance_details) as data_file:
+            data = json.load(data_file)
+    except json.JSONDecodeError as e:
+        print("Error when parsing pricing information: {0}".format(e.msg))
+        raise
+
     instances = {}
-    for sku, product in data.get("products").iteritems():
+    for sku, product in data.get("products").items():
         if "Compute Instance" in product.get("productFamily"):
             instance = product.get("attributes")
             instances[instance.get("instanceType")] = {"vcpus": instance.get("vcpu")}
+            # Adding instance's gpu information to instances.json
+            if "gpu" in instance:
+                instances[instance.get("instanceType")]["gpu"] = instance.get("gpu")
+
+    # Check if all info from old instances.json is included in the new instances.json
+    validate_against_old_version(instances, region, aws_credentials)
+
     print(json.dumps(instances))
     json.dump(instances, open("instances.json", "w"))
 
 
+def validate_against_old_version(instances_info, region, aws_credentials=None):
```

Comment:
what about comparing with something as simple as?
```
dict(new, **old) == new
```

This should also verifies the value linked to every nested key

Reply:
The comparison you suggested is definitely simpler and works just as well, but I would argue that the double for loop helps with debugging because we would know right away which instance type of resource is missing.

Reply:
agreed but at the same time we are not checking the values associated to resources keys and this test needs to be updated if we modify the file. 

printing the two files (even better if sorted) should be a good way to debug failures